components/src/BulkOrder/bulkorder.main.tsx

```diff
@@ -216,26 +218,26 @@ export class BulkOrder extends React.Component<BulkOrderProps, BulkOrderState> {
   }
 
   handleBarcodeScanned(barcode) {
-    const { items } = this.state;
-    console.warn(barcode, items);
     login().then(() => {
       searchLookup(barcode)
         .then((res) => {
           const foundItem = res._element[0];
-          const emptyItem = items.find(item => item.code === '');
-          const updatedItems = items.map((item) => {
-            if (item.key === emptyItem.key) {
-              return {
-                ...item,
-                code: foundItem._code[0].code,
-                product: foundItem,
-              };
-            }
+          this.setState((state) => {
+            const emptyItem = state.items.find(item => item.code === '');
+            const updatedItems = state.items.map((item) => {
+              if (item.key === emptyItem.key) {
+                return {
+                  ...item,
+                  code: foundItem._code[0].code,
```

Comment:
This is fetching the code for the product, not the code for the sku (which is what we need). Below you can see the response contains the product code (first), then the sku code (nested in response):
![image](https://user-images.githubusercontent.com/39800563/62728252-4e149100-b9e9-11e9-9fbd-754eb1b48833.png)